.circleci/config.yml

```diff
@@ -278,43 +389,19 @@ jobs:
   checkout:
     executor: basic
     steps:
-      - restore_cache:
-          keys:
-            - source-{{ .Branch }}-{{ .Revision }}
-            - source-{{ .Branch }}
-            - source-
       - checkout
-      - save_cache:
-          key: source-{{ .Branch }}-{{ .Revision }}
-          paths:
-            - .git
-      # The basic idea here is to generate a file with checksums for all the backend source files, and save it as
-      # `./backend-checksums.txt`. Then we'll use the checksum of those files for uberjar caching and for determining
-      # whether we need to run driver tests again; thus we can reuse the same uberjar for integration tests across any
-      # build where the backend files are the same
-      - run:
-          name: Generate checksums of all backend source files to use as cache key
-          command: >
-            for file in `find . -type f -name '*.clj' | sort`;
-              do echo `md5sum $file` >> backend-checksums.txt;
-            done;
-            echo `md5sum project.clj` >> backend-checksums.txt
-      # Do the same for the frontend
-      - run:
-          name: Generate checksums of all frontend source files to use as Uberjar cache key
-          command: >
-            for file in `find ./frontend ./enterpise/frontend -type f | sort`;
-              do echo `md5sum $file` >> frontend-checksums.txt;
-            done;
-            echo `md5sum yarn.lock` >> frontend-checksums.txt
-            echo `md5sum webpack.config.js` >> frontend-checksums.txt
-      # As well as driver modules (database drivers) -- used only for caching the built drivers
-      - run:
-          name: Generate checksums of all driver module source files to use as cache key
-          command: >
-            for file in `find ./modules -type f -name '*.clj' -or -name '*.yaml' | sort`;
-              do echo `md5sum $file` >> modules-checksums.txt;
-            done;
+      # .BACKEND-CHECKSUMS is every Clojure source file as well as dependency files like deps.edn and plugin manifests
+      - create-checksum-file:
+          filename: .BACKEND-CHECKSUMS
+          find-args: ". -type f -name '*.clj' -or -name '*.java' -or -name deps.edn -or -name metabase-plugin.yaml"
+      # .FRONTEND-CHECKSUMS is every JavaScript source file as well as dependency files like yarn.lock
+      - create-checksum-file:
+          filename: .FRONTEND-CHECKSUMS
+          find-args: ". -type f -name '*.js' -or -name '*.jsx' -or -name yarn.lock -or -name webpack.config.js -or -name package.json"
```

Comment:
How about the assets? `resources/frontend_client/app/assets/geojson/`. Is it already covered here?

Reply:
maybe it makes sense to change this to include all JSON files as well?

Reply:
went ahead and added all JSON files as well